src/bootstrap/builder.rs

```diff
@@ -586,6 +586,29 @@ impl<'a> Builder<'a> {
             extra_args.push_str(&s);
         }
 
+        let mut target_features = Vec::new();
```

Comment:
I would prefer to keep this logic inside `rustc.rs` instead of through RUSTFLAGS.

Reply:
As noted in the commit message some tools are built with a compiler requirement of stage = 0 and build = host, which causes them not to go through rustc.rs:

https://github.com/rust-lang/rust/blob/25749ad66de71ffb46e6b04fb0cccf1ae4fc5ed5/src/bootstrap/builder.rs#L509-L515
https://github.com/rust-lang/rust/blob/25749ad66de71ffb46e6b04fb0cccf1ae4fc5ed5/src/bootstrap/lib.rs#L1201-L1203

When deciding how to fix it (I could as well try to make it go through rustc.rs even for stage = 0 and build = host, though my first attempt at it broke the build completely) I was going by the comment at the start of rustc.rs that suggests this is the way forward for some parts of rustc.rs:

https://github.com/rust-lang/rust/blob/25749ad66de71ffb46e6b04fb0cccf1ae4fc5ed5/src/bootstrap/bin/rustc.rs#L24-L26

Reply:
I believe all invocations of rustc are intended to go through `rustc.rs` -- the code you point to doesn't quite indicate to me that rustc is being invoked directly. I personally prefer using the rustc shim over rustflags when possible since it's easier to read and interpret (at least for me).

Reply:
It seems I was misreading the code and you are right. The result of the function I linked seems to mainly be used to set the environment variable `RUSTC_REAL`, while setting `RUSTC` to the shim.

This leaves only the logic in `rustc.rs` that disables most of the flag passing if `--target` wasn't passed as culprit for my build failure.

Reply:
This should be done properly now in the newest version this pull request.